### PR TITLE
Add movable aa1 to improve basics aa3

### PR DESCRIPTION
**Pull request by schettn**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Documentation (non-breaking change which adds documentation)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently a user cannot modify the position of charts on his own overview page. #88 

**What is the new behavior (if this is a feature change)?**
- Now a user can modify the position of charts on his own overview page. #88

**Other information**:
- 🐍

### CODE REVIEW

File: src/components/molecules/MovableBoundary/index.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
+//#region > Imports
+//> React
+// Contains all the functionality necessary to define React components
+import React, { lazy, Suspense } from "react";
+// DOM bindings for React Router
+import { Link, withRouter } from "react-router-dom";
+// React PropTypes
+import PropTypes from "prop-types";
+
```
Comment (pinterid) [inline src/components/molecules/MovableBoundary/index.jsx]:
```suggestion
```

File: src/components/molecules/MovableBoundary/index.jsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+//> MDB Sortable
+// MDB plugin for sortable item lists
+import MDBSortable from "mdb-react-sortable";
+
+//> SCSS
```
Comment (pinterid) [inline src/components/molecules/MovableBoundary/index.jsx]: All the other files used to have CSS and not SCSS, is this a new standard or a mistake?
```suggestion
//> CSS
```

File: src/components/molecules/MovableBoundary/index.jsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        return (
+          <div className="SortableItem unsortable" disabled>
+            {item}
+          </div>
+        );
+      }
+    });
+  }
+
+  // Swaps indexArray position of element.
```
Comment (pinterid) [inline src/components/molecules/MovableBoundary/index.jsx]:
```suggestion
  // Swaps indexArray position of element
```

File: src/components/molecules/MovableBoundary/index.jsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+    return orderedItems;
+  }
+
+  // Store indexArray.
+  saveItemOrder() {
+    this.props.pool[this.props.uid] = JSON.stringify(this.state.indexArray);
+  }
+
+  // Returns items to be rendered in the Sortable.
```
Comment (pinterid) [inline src/components/molecules/MovableBoundary/index.jsx]:
```suggestion
  // Returns items to be rendered in the Sortable
```

File: src/components/molecules/MovableBoundary/index.jsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+      indexArray = baseIndexArray;
+    } else {
+      indexArray = JSON.parse(indexArray);
+    }
+
+    this.setState({ indexArray });
+  }
+
+  // Returns items ordered by indexArray.
```
Comment (pinterid) [inline src/components/molecules/MovableBoundary/index.jsx]:
```suggestion
  // Returns items ordered by indexArray
```

File: src/components/molecules/MovableBoundary/index.jsx [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    });
+
+    // If props.edit changes to false, save indexArray.
+    if (!nextProps.edit && this.props.edit) {
+      this.saveItemOrder();
+    }
+  };
+
+  componentDidMount() {
+    // Load item order after mounting.
```
Comment (pinterid) [inline src/components/molecules/MovableBoundary/index.jsx]:
```suggestion
    // Load item order after mounting
```

File: src/components/molecules/MovableBoundary/index.jsx [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    items: this.props.items ? this.props.items : null,
+    indexArray: null,
+  };
+
+  componentWillReceiveProps = (nextProps) => {
+    this.setState({
+      items: nextProps.items ? nextProps.items : null,
+    });
+
+    // If props.edit changes to false, save indexArray.
```
Comment (pinterid) [inline src/components/molecules/MovableBoundary/index.jsx]:
```suggestion
    // If props.edit changes to false, save indexArray
```

File: src/components/molecules/MovableBoundary/index.jsx [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+// MDB plugin for sortable item lists
+import MDBSortable from "mdb-react-sortable";
+
+//> SCSS
+import "./movableboundary.scss";
+//#endregion
+
+//#region > Components
+/**
+ * @class This enables implememntation of horizontally or vertically movable items into a page.
```
Comment (pinterid) [inline src/components/molecules/MovableBoundary/index.jsx]:
```suggestion
 * @class This enables implememntation of horizontally or vertically movable items into a page
```

File: src/components/molecules/MovableBoundary/index.jsx [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+    let orderedItems = [];
+
+    this.state.indexArray.map((index) => {
+      orderedItems = [...orderedItems, this.state.items[index]];
+    });
+
+    return orderedItems;
+  }
+
+  // Store indexArray.
```
Comment (pinterid) [inline src/components/molecules/MovableBoundary/index.jsx]:
```suggestion
  // Store indexArray
```

File: src/components/molecules/MovableBoundary/index.jsx [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+          <>
+            <MDBSortable
+              axis={this.props.movementAxis}
+              items={this.renderItems(this.props.edit ? true : false)}
+              itemClassName={
+                this.props.edit ? "SortableItem" : "SortableItem unsortable"
+              }
+              listClassName="SortableList"
+              onSortEnd={(obj, e) => {
+                // When item is let go of, change indexArray position of Item.
```
Comment (pinterid) [inline src/components/molecules/MovableBoundary/index.jsx]:
```suggestion
                // When item is let go of, change indexArray position of Item
```

File: src/components/organisms/tabs/OverviewTab/index.jsx [hunk 11]
```diff
@@ -19,6 +23,8 @@ import {
   LatestActivity,
   ErrorBoundary,
 } from "../../../atoms";
+import { MovableBoundary } from "../../../molecules";
+
```
Comment (pinterid) [inline src/components/organisms/tabs/OverviewTab/index.jsx]:
```suggestion
```

File: src/components/organisms/tabs/OverviewTab/index.jsx [hunk 12]
```diff
@@ -66,127 +73,193 @@ class OverviewTab extends React.Component {
     });
   };
 
+  componentDidMount = () => {};
```
Comment (pinterid) [inline src/components/organisms/tabs/OverviewTab/index.jsx]: Can this be removed?
```suggestion
```

File: src/components/organisms/tabs/OverviewTab/index.jsx [hunk 13]
```diff
@@ -66,127 +73,193 @@ class OverviewTab extends React.Component {
...
+
+  handleEditClick = (platformData) => {
+    if (this.state.edit) {
+      this.props.writeCache(platformData);
+    }
```
Comment (pinterid) [inline src/components/organisms/tabs/OverviewTab/index.jsx:78]:
```suggestion
    }
    
```

File: src/components/organisms/tabs/OverviewTab/index.jsx [hunk 14]
```diff
@@ -66,127 +73,193 @@ class OverviewTab extends React.Component {
...
+    // Create empty pool if there isn't already one
+    if (!fetchedUser.platformData.user.movablePool) {
+      fetchedUser.platformData.user.movablePool = {};
+    }
+
+    const movablePool = fetchedUser.platformData.user.movablePool;
 
     return (
       <>
         {/* <ErrorBoundary hidden="true"> */}
```
Comment (pinterid) [inline src/components/organisms/tabs/OverviewTab/index.jsx:96]: Can this be removed.
```suggestion
```

**Review by pinterid — CHANGES_REQUESTED** [review]
Found some mistakes.

File: src/components/organisms/tabs/OverviewTab/index.jsx (hunk 14 above)
Comment (schettn) [reply src/components/organisms/tabs/OverviewTab/index.jsx:96]: Nep

**Review by pinterid — APPROVED** [review]
Looks good.

**Review by Aichnerc — APPROVED** [review]
LGTM 🕳

**Pull request merged**